### DISCUSSION TITLE
Immutable attribute need to use "get" prefix?

### DISCUSSION THREAD

**Discussion by pandoras-toolbox** [mapstruct/mapstruct#3641] [General]
Today I stumbled over a basic MapStruct behavior. Actually I think the reason is said in this StackOverflow posting because I had the same error message: https://stackoverflow.com/a/62846707

Does that mean this method name style is not supported by MapStruct?

```
@Value.Style(get = "*", set = "*")
@Value.Immutable
public interface Abc {
    String myAttribute();
}
```

Does it really has to be like that?

```
@Value.Immutable
public interface Abc {
    String getMyAttribute();
}
```

It worked for me finally with the second variant, not with the first variant. But I would like to have the freedom not to use "get" prefix for the immutables attribute names and still be able to use MapStruct if possible?

**Comment by thunderhook** [comment]
Yes, MapStruct is looking for accessors based on the JavaBeans conventions, see here:
* https://mapstruct.org/documentation/stable/reference/html/#basic-mappings
* JavaBeans Spec: https://www.oracle.com/java/technologies/javase/javabeans-spec.html

You can provide your own [`AccessorNamingStrategy`](https://github.com/mapstruct/mapstruct/blob/main/processor/src/main/java/org/mapstruct/ap/spi/AccessorNamingStrategy.java) if you really need to support something like this. See:
* https://mapstruct.org/documentation/stable/reference/html/#_custom_accessor_naming_strategy

**Comment by pandoras-toolbox** [comment]
Thank you!

It surprised me especially because in the previous company I used also Immutables and MapStruct with getter/setters not having any prefix, and it worked.

I think it is more easy to read without "get"/"set" prefixes, especially when setting many attributes.

Therefore I think I will try that `AccessorNamingStrategy` out, to see if I can support that.

**Reply by filiphr** [reply]
Perhaps your previous company had a custom `AccessorNamingStrategy` that made this work.

By the way this is directly linked to #1924. We do have a PR #3189 for this, but we haven't had the time to properly test it and integrate it into MapStruct. Perhaps something to look into for 1.7
